lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -476,30 +476,31 @@ struct AIEObjectFifoStatefulTransformPass
         target = objFifoLinks[*linkOp];
 
     // search for MemOp
-    MemOp *producerMem = nullptr;
+    Operation *producerMem = nullptr;
     for (auto memOp : device.getOps<MemOp>()) {
       if (memOp.getTile() == op.getProducerTile()) {
-        producerMem = &memOp;
+        producerMem = memOp.getOperation();
         break;
       }
     }
 
     // if none exists, create one
     TileOp objFifoTileOp = target.getProducerTileOp();
     if (producerMem == nullptr) {
+      if (device->getNumRegions() != 1)
+        assert(false && "expected num regions for device op");
+      OpBuilder::InsertionGuard g(builder);
       builder.setInsertionPointToEnd(device.getBody());
       MemOp newMemOp =
           builder.create<MemOp>(builder.getUnknownLoc(), objFifoTileOp);
-      producerMem = &newMemOp;
-      Region &r = producerMem->getBody();
-      r.push_back(new Block);
-      // add terminator operation to end block
-      Block &endBlock = r.back();
-      builder.setInsertionPointToStart(&endBlock);
-      builder.create<EndOp>(builder.getUnknownLoc());
```

Comment:
Some wacky things going on here that were giving me assert fails because the `MemOp *` kept being set to point at a `AIE_TileOp` which has no regions (and thus would then fail the assert in `producerMem->getBody()`).